# 1. Что покажет приведенный ниже фрагмент кода?
```
s = 'abcdefg'
print(s[0] + s[2] + s[4] + s[6])
```

Ответ: `aceg`

# 2. Что покажет приведенный ниже фрагмент кода?
```
s = 'abcdefg'
print(s[0]*3 + s[-1]*3 + s[3]*2 + s[3]*2)
```

Ответ: `aaagggdddd`


# 3. Что покажет приведенный ниже фрагмент кода?
```
s = '01234567891011121314151617'
for i in range(0, len(s), 5):
    print(s[i], end='')
```

Ответ: `051217`


# 4. Дополните приведенный код, используя индексатор, так чтобы он вывел символ запятой.

In [ ]:
s = "In 2010, someone paid 10k Bitcoin for two pizzas."
print(s[7])

,


# 5. Дополните приведенный код, используя индексатор, так чтобы он вывел символ w.

In [ ]:
s = "In 2010, someone paid 10k Bitcoin for two pizzas."
print(s[-10])

w


# 6. В столбик 1
На вход программе подается одна строка. Напишите программу, которая выводит элементы строки с индексами 0, 2, 4, ... в столбик.

`Формат входных данных`

На вход программе подается одна строка.

`Формат выходных данных`

Программа должна вывести элементы строки с индексами 0, 2, 4, ..., каждое на отдельной строке.

In [1]:
s = input()
for i in range(0, len(s), 2):
    print(s[i])

abcdefghij
a
c
e
g
i


# 7. В столбик 2
На вход программе подается одна строка. Напишите программу, которая выводит в столбик элементы строки в обратном порядке.

`Формат входных данных`

На вход программе подается одна строка.

`Формат выходных данных`

Программа должна вывести в столбик элементы строки в обратном порядке. 

In [2]:
s = input()
for i in range(len(s) - 1, -1, -1):
    print(s[i])

abc
c
b
a


# 8. ФИО
На вход программе подаются три строки: имя, фамилия и отчество. Напишите программу, которая выводит инициалы человека.

`Формат входных данных`

На вход программе подаются три строки, каждая на отдельной строке.

`Формат выходных данных`

Программа должна вывести ФИО человека.

Примечание. Гарантируется, что имя, фамилия и отчество начинаются с заглавной буквы.

In [3]:
s, n, sn = input(), input(), input()
print(n[0], s[0], sn[0], sep='')

Тимур
Гуев
Ахсарбекович
ГТА


# 9. Цифра 1
На вход программе подается одна строка состоящая из цифр. Напишите программу, которая считает сумму цифр данной строки.

`Формат входных данных`

На вход программе подается одна строка состоящая из цифр.

`Формат выходных данных`

Программа должна вывести сумму цифр данной строки.

In [4]:
s = input()
total = 0
for i in range(len(s)):
    total += int(s[i])
print(total)

2514
12


# 10. Цифра 2
На вход программе подается одна строка. Напишите программу, которая выводит сообщение «Цифра» (без кавычек), если строка содержит цифру. В противном случае вывести сообщение «Цифр нет» (без кавычек).

`Формат входных данных`

На вход программе подается одна строка.

`Формат выходных данных`

Программа должна вывести текст в соответствии с условием задачи.

In [7]:
n = input()
s = 'Цифр нет'
for i in range(len(n)):
    if n[i] in '0123456789':
        s = 'Цифра'
print(s)

Hi 17 Python
Цифра


# 11. Сколько раз?
На вход программе подается одна строка. Напишите программу, которая определяет сколько раз в строке встречаются символы + и *.

`Формат входных данных`

На вход программе подается одна строка.

`Формат выходных данных`

Программа должна вывести сколько раз встречаются символы  + и * в строке.

In [8]:
s = input()
pl = 0
zv = 0
for i in range(len(s)):
    if s[i] == '+':
        pl += 1
    if s[i] == '*':
        zv += 1
print('Символ + встречается', pl, 'раз')
print('Символ * встречается', zv, 'раз')


bcd+a++++**31415
Символ + встречается 5 раз
Символ * встречается 2 раз


# 12. Одинаковые соседи
На вход программе подается одна строка. Напишите программу, которая определяет сколько в ней одинаковых соседних символов.

`Формат входных данных`

На вход программе подается одна строка.

`Формат выходных данных`

Программа должна вывести количество одинаковых соседних символов.

In [9]:
s = input()
total = 0
for i in range(0, len(s) - 1):
    if s[i] == s[i+1]:
        total += 1
print(total)

aaa
2


# 13. Гласные и согласные
На вход программе подается одна строка с буквами русского языка. Напишите программу, которая определяет количество гласных и согласных букв.

`Формат входных данных`

На вход программе подается одна строка.

`Формат выходных данных`

Программа должна вывести количество гласных и согласных букв.

Примечание. В русском языке 10 гласных букв (а, у, о, ы, и, э, я, ю, ё, е) и 21 согласная буква (б, в, г, д, ж, з, й, к, л, м, н, п, р, с, т, ф, х, ц, ч, ш, щ).

In [10]:
s = input()
gl = 0
sog = 0
for i in range(0, len(s)):
    if s[i] in 'ауоыиэяюёеАУОЫИЭЯЮЁЕ':
        gl += 1
    if s[i] in 'бвгджзйклмнпрстфхцчшщБВГДЖЗЙКЛМНПРСТФХЦЧШЩ':
        sog += 1
print('Количество гласных букв равно', gl)
print('Количество согласных букв равно', sog)

Вдохновение – это умение приводить себя в рабочее состояние!
Количество гласных букв равно 25
Количество согласных букв равно 24


# 14. Decimal to Binary
На вход программе подается натуральное число, записанное в десятичной системе счисления. Напишите программу, которая переводит данное число в двоичную систему счисления.

`Формат входных данных`

На вход программе подается одно натуральное число.

`Формат выходных данных`

Программа должна вывести число записанное в двоичной системе счисления.

In [11]:
n = int(input()) 
d = ''          
while n > 0:
    d = str(n % 2) + d
    n //= 2
print(d)


128
10000000


# 15. Что покажет приведенный ниже фрагмент кода?
```
s = 'abcdefg'
print(s[2:5])
```

Ответ: `cde`

# 16. Что покажет приведенный ниже фрагмент кода?
```
s = 'abcdefg'
print(s[3:])
```

Ответ: `defg`

# 17. Что покажет приведенный ниже фрагмент кода?
```
s = 'abcdefg'
print(s[:3])
```

Ответ: `abc`


# 18. Что покажет приведенный ниже фрагмент кода?
```
s = 'abcdefg'
print(s[:])
```

Ответ: `abcdefg`


# 19. Что покажет приведенный ниже фрагмент кода?
```
s = 'abcdefg'
print(s[::-3])
```

Ответ: `gda`


# 20. Дополните приведенный код, используя срезы, так чтобы он вывел первые 12 символов строки s.

In [ ]:
s = "In 2010, someone paid 10k Bitcoin for two pizzas."
print(s[:12])

In 2010, som


# 21. Дополните приведенный код, используя срезы, так чтобы он вывел последние 9 символов строки s.

In [ ]:
s = "In 2010, someone paid 10k Bitcoin for two pizzas."
print(s[-9:])

o pizzas.


# 22. Дополните приведенный код, используя срезы, так чтобы он вывел каждый 7 символ строки s начиная от начала строки.

In [ ]:
s = "In 2010, someone paid 10k Bitcoin for two pizzas."
print(s[::7])

I,n top


# 23. Дополните приведенный код, используя срезы, так чтобы он вывел строку s в обратном порядке.

In [ ]:
s = "In 2010, someone paid 10k Bitcoin for two pizzas."
print(s[::-1])

.sazzip owt rof nioctiB k01 diap enoemos ,0102 nI


# 24. Палиндром
На вход программе подается одно слово, записанное в нижнем регистре. Напишите программу, которая определяет является ли оно палиндромом.

``Формат входных данных``

На вход программе подается одно слово в нижнем регистре.

``Формат выходных данных``

Программа должна вывести «YES», если слово является палиндромом и «NO» в противном случае.

Примечание. Палиндром читается одинаково в обоих направлениях, например слово «потоп».

In [12]:
s = input()
if s[::] == s[::-1]:
    print('YES')
else:
    print('NO')

потоп
YES


# 25. Делаем срезы 1
На вход программе подается одна строка. Напишите программу, которая выводит:

* общее количество символов в строке;
* исходную строку повторенную 3 раза;
* первый символ строки;
* первые три символа строки;
* последние три символа строки;
* строку в обратном порядке;
* строку с удаленным первым и последним символом.

`Формат входных данных`

На вход программе подается одна строка, длина которой больше 3 символов.

`Формат выходных данных`

Программа должна вывести данные в соответствии с условием. Каждое значение выводится на отдельной строке.

In [13]:
s = input()
print(len(s))
print(s[:] * 3)
print(s[0])
print(s[:3])
print(s[-3:])
print(s[::-1])
print(s[1:-1])

abcdefghijklmnopqrstuvwxyz
26
abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz
a
abc
xyz
zyxwvutsrqponmlkjihgfedcba
bcdefghijklmnopqrstuvwxy


# 26. Делаем срезы 2
На вход программе подается одна строка. Напишите программу, которая выводит:

* третий символ этой строки;
* предпоследний символ этой строки;
* первые пять символов этой строки;
* всю строку, кроме последних двух символов;
* все символы с четными индексами;
* все символы с нечетными индексами;
* все символы в обратном порядке;
* все символы строки через один в обратном порядке, начиная с последнего.

`Формат входных данных`
На вход программе подается одна строка, длина которой больше 5 символов.

`Формат выходных данных`

Программа должна вывести данные в соответствии с условием. Каждое значение выводится на отдельной строке.

In [14]:
s = input()
print(s[2])
print(s[-2])
print(s[:5])
print(s[:-2])
print(s[::2])
print(s[1::2])
print(s[::-1])
print(s[-1::-2])

abcdefghijklmnopqrstuvwxyz
c
y
abcde
abcdefghijklmnopqrstuvwx
acegikmoqsuwy
bdfhjlnprtvxz
zyxwvutsrqponmlkjihgfedcba
zxvtrpnljhfdb


# 27. Две половинки
На вход программе подается строка текста. Напишите программу, которая разрежет ее на две равные части, переставит их местами и выведет на экран.

`Формат входных данных`

На вход программе подается строка текста.

`Формат выходных данных`

Программа должна вывести текст в соответствии с условием задачи.

Примечание. Если длина строки нечетная, то длина первой части должна быть на один символ больше.

In [15]:
s = input()
x = len(s)
a = x // 2 + x % 2
print(s[a:] + s[:a])

efgabcd
bcdefga


# 28. Что покажет приведенный ниже фрагмент кода?
```
s = 'i Learn Python language'
print(s.capitalize())
```

Ответ: `I learn python language`

# 29. Что покажет приведенный ниже фрагмент кода?
```
s = 'i LEARN Python LAnguaGE'
print(s.lower())
```

Ответ: `i learn python language`

# 30. Что покажет приведенный ниже фрагмент кода?
```
s = '$12344%^$#@!'
print(s.lower())
```

Ответ: `$12344%^$#@!`

# 31. Что покажет приведенный ниже фрагмент кода?
```
s1 = 'a'
s2 = s1.upper()
print(s1, s2)
```

Ответ: `a A`

# 32. Что покажет приведенный ниже фрагмент кода?
```
s = 'i LEARN Python LAnguaGE'
print(s.upper())
```

Ответ: `I LEARN PYTHON LANGUAGE`

# 33. Что покажет приведенный ниже фрагмент кода?
```
s = 'i LEARN Python LAnguaGE'
print(s.swapcase())
```

Ответ: `I learn pYTHON laNGUAge`

# 34. Заглавные буквы
На вход программе подается строка состоящая из имени и фамилии человека, разделенных одним пробелом. Напишите программу, которая проверяет, что имя и фамилия начинаются с заглавной буквы.

`Формат входных данных` 

На вход программе подается строка.

`Формат выходных данных`

Программа должна вывести «YES» если имя и фамилия начинаются с заглавной буквы и «NO» в противном случае.

Примечание. Строка содержит только буквы. 

In [16]:
s = input()
if s == s.title():
    print('YES')
else:
    print('NO')

chris alan
NO


# 35. sWAP cASE
На вход программе подается строка. Напишите программу, которая меняет регистр символов, другими словами замените все строчные символы заглавными и наоборот.

`Формат входных данных`

На вход программе подается строка.

`Формат выходных данных`

Программа должна вывести строку в соответствии с условием задачи.

In [17]:
n = input()
print(n.swapcase())

Pythonist 2
pYTHONIST 2


# 36. Хороший оттенок
На вход программе подается строка текста. Напишите программу, которая определяет является ли оттенок текста хорошим или нет. Текст имеет хороший оттенок, если содержит подстроку «хорош» во всевозможных регистрах.

`Формат входных данных `

На вход программе подается строка текста.

`Формат выходных данных`

Программа должна вывести «YES» если текст имеет хороший оттенок и «NO» в противном случае.

Примечание. Текст содержащий хорош, ХОРОШ, Хорош, хОРОШ и т.д. имеет хороший оттенок.

In [18]:
s = input()
s = s.lower()
if 'хорош' in s:
    print('YES')
else:
    print('NO')

я очень хороший текст =)
YES


# 37. Нижний регистр
На вход программе подается строка. Напишите программу, которая подсчитывает количество буквенных символов в нижнем регистре.

`Формат входных данных `

На вход программе подается строка.

`Формат выходных данных`

Программа должна вывести количество буквенных символов в нижнем регистре.

In [20]:
s = input()
total = 0
for i in range(0, len(s)):
    if s[i] in 'abcdefghijklmnopqrstuvwxyz':
        total += 1
print(total)

abcABCD12345
3


# 38. Что покажет приведенный ниже фрагмент кода?
```
s = 'aabbAAccDDaa'
s = s.lower()
print(s.count('a'))
```

Ответ: `6`

# 39. Что покажет приведенный ниже фрагмент кода?
```
s = 'www.stepik.org'
print(s.startswith('www'))
```

Ответ: `True`

# 40. Что покажет приведенный ниже фрагмент кода?
```
s = 'www.stepik.org'
print(s.endswith('.ru'))
```

Ответ: `False`


# 41. Что покажет приведенный ниже фрагмент кода?
```
s = 'I learn Python language. Python - awesome!'
print(s.find('Python'))
```

Ответ: `8`


# 42. Что покажет приведенный ниже фрагмент кода?
```
s = '     I learn Python language               '
print(s.strip())
```

Ответ: `I learn Python language`

# 43. Что покажет приведенный ниже фрагмент кода?
```
s = 'abcdababa'
print(s.replace('ab', '123'))
```

Ответ: `123cd123123a`

# 44. Количество слов
На вход программе подается строка текста, состоящая из слов, разделенных ровно одним пробелом. Напишите программу, которая подсчитывает количество слов в ней.

`Формат входных данных `

На вход программе подается строка текста.

`Формат выходных данных`

Программа должна вывести количество слов.

Примечание 1. Строка текста не содержит пробелов в начале и конце.

Примечание 2. Используйте для решения задачи метод count.

In [21]:
s = input()
print(int(s.count(' ')) + 1)

In 2010, someone paid 10k Bitcoin for two pizzas.
9


# 45. Минутка генетики
На вход программе подается строка генетического кода, состоящая из букв А (аденин), Г (гуанин), Ц (цитозин), Т (тимин). Напишите программу, которая подсчитывает сколько аденина, гуанина, цитозина и тимина входит в данную строку генетического кода.

`Формат входных данных `

На вход программе подается строка генетического кода, состоящая из символов А, Г, Ц, Т, а, г, ц, т.

`Формат выходных данных`

Программа должна вывести сколько гуанина, тимина, цитозина, аденина входит в данную строку генетического кода.

Примечание. Строка не содержит символов, кроме как А, Г, Ц, Т, а, г, ц, т.

In [22]:
s = input()
s = s.lower()
print('Аденин:', s.count('а'))
print('Гуанин:', s.count('г'))
print('Цитозин:', s.count('ц'))
print('Тимин:', s.count('т'))

ааггццттААГГЦЦТТ
Аденин: 4
Гуанин: 4
Цитозин: 4
Тимин: 4


# 46. Очень странные дела
Джим Хоппер с помощью радиоприемника пытается получить сообщение Оди. На приемник ему поступает nn различных последовательностей кода Морзе. Декодировав их, он получает последовательности из цифр и строчного латинского алфавита, при этом во всех сообщениях Оди содержится число 11, причем минимум 3 раза. Помогите определить Джиму количество сообщений от Оди.

`Формат входных данных`

В первой строке подаётся число nn – количество сообщений, в последующих nn строках вводятся строки, содержащие латинские строчные буквы и цифры.

`Формат выходных данных`

Программа должна вывести количество строк в которых содержится число 11 минимум 3 раза.

Примечание: Числа 11 необязательно должны быть разделены другими символами, нужно подсчитать вхождение последовательности символов "11", т.е. например в строке "111" содержится одна такая последовательность, в то время как в "1111" их уже две.

In [25]:
n = int(input())
tot = 0
for i in range(n):
    i = input()
    if i.count('11') >= 3:
        tot += 1
print(tot)

3
58vfsfgsdf2
69sdfsdfg47sdf
235
0


# 47. Количество цифр
На вход программе подается строка текста. Напишите программу, которая подсчитывает количество цифр в данной строке.

`Формат входных данных `

На вход программе подается строка текста.

`Формат выходных данных`

Программа должна вывести количество цифр в данной строке.

In [26]:
s = input()
tot = 0
for i in range(10):
    tot += s.count(str(i))
print(tot)

l33t 3301
6


# 48. .com or .ru
На вход программе подается строка текста. Напишите программу, которая проверяет, что строка заканчивается подстрокой .com или .ru.

`Формат входных данных `

На вход программе подается строка текста.

`Формат выходных данных`

Программа должна вывести «YES» если введенная строка заканчивается подстрокой .com или .ru и «NO» в противном случае.

In [27]:
s = input()
if s.endswith('.ru') == True or s.endswith('.com') == True:
    print('YES')
else:
    print('NO')

www.yandex.ru
YES


# 49. Самый частотный символ
На вход программе подается строка текста. Напишите программу, которая выводит на экран символ, который появляется наиболее часто.

`Формат входных данных `

На вход программе подается строка текста. Текст может содержать строчные и заглавные буквы английского и русского алфавита, а также цифры.

`Формат выходных данных`

Программа должна вывести символ, который появляется наиболее часто.

Примечание 1. Если таких символов несколько, следует вывести последний по порядку символ.

Примечание 2. Следует различать заглавные и строчные буквы, а также буквы русского и английского алфавита.

In [28]:
s, a, b = input(), 0, 0
for i in s:
    if s.count(i) >= a:
        a = s.count(i)
        b = i
print(b)

abaabbcccc
c


# 50. Первое и последнее вхождение
На вход программе подается строка текста. Если в этой строке буква «f» встречается только один раз, выведите её индекс. Если она встречается два и более раз, выведите индекс её первого и последнего вхождения на одной строке, разделенных символом пробела. Если буква «f» в данной строке не встречается, следует вывести «NO».

`Формат входных данных `

На вход программе подается строка текста.

`Формат выходных данных`

Программа должна вывести текст в соответствии с условием задачи.

In [29]:
s = input()
if s.count('f') == 1:
    print(s.find('f'))
elif s.count('f') > 1:
    print(s.find('f'), s.rfind('f'))
else:
    print('NO')

abcdef
5


# 51. Удаление фрагмента
На вход программе подается строка текста, в которой буква «h» встречается минимум два раза. Напишите программу, которая удаляет из этой строки первое и последнее вхождение буквы «h», а также все символы, находящиеся между ними.

`Формат входных данных `

На вход программе подается строка текста.

`Формат выходных данных`

Программа должна вывести текст в соответствии с условием задачи.

In [30]:
s = input()
print(s[:s.find('h')], s[s.rfind('h') + 1:], sep='')

ahahahahaha
aa


# 52. Что покажет приведенный ниже фрагмент кода?
```
s = 'aabbAA111ccDDaa'
print(s.isalnum())
print(s.isalpha())
print(s.isdigit())
```


* True
* False
* False

```
True
True
False
```
```
False
False
False
```
```
True
True
True
```

# 53. Что покажет приведенный ниже фрагмент кода?
```
s = 'aabb!@#$11cc'
print(s.islower())
```


* False

* **True**

# 54. Что покажет приведенный ниже фрагмент кода?
```
s = 'AAb!@#$11CC'
print(s.isupper())
```


* **False**

* True

# 55. Что покажет приведенный ниже фрагмент кода?
```
s = '    abbc    '
print(s.isspace())
```


* **False**

* True

# 56. Дополните приведенный код, используя форматирование строк с помощью метода format, так чтобы он вывел текст: 

«In 2010, someone paid 10k Bitcoin for two pizzas.» (без кавычек).

In [31]:
s = 'In {0}, someone paid {1} {2} for two pizzas.'
a = 2010
b = '10k'
c = 'Bitcoin'
print(s.format(a, b, c))

In 2010, someone paid 10k Bitcoin for two pizzas.


# 57. Дополните приведенный код, используя форматирование строк с помощью f-строк, так чтобы он вывел текст: 

«In 2010, someone paid 10K Bitcoin for two pizzas.» (без кавычек).

In [32]:
year = 2010
amount = '10K'
currency = 'Bitcoin'
print(f'In {year}, someone paid {amount} {currency} for two pizzas.')

In 2010, someone paid 10K Bitcoin for two pizzas.


# 58. Что покажет приведенный ниже фрагмент кода?
```
print(ord('foo'))
```


* **Произойдет ошибка TypeError**

* 324

* 102 111 111

* 102

# 59. Символы в диапазоне
На вход программе подаются два числа a и b. Напишите программу, которая для каждого кодового значения в диапазоне от a до b (включительно), выводит соответствующий ему символ из таблицы символов Unicode.

`Формат входных данных` 

На вход программе подается два натуральных числа, каждое на отдельное строке.

`Формат выходных данных`

Программа должна вывести текст в соответствии с условием задачи.

In [33]:
a, b = int(input()), int(input())
for i in range(a, b + 1):
    print(chr(i), end=' ')

65
70
A B C D E F 

# 60. Простой шифр
На вход программе подается строка текста. Напишите программу, которая переводит каждый ее символ в соответствующий ему код из таблицы символов Unicode.

`Формат входных данных` 

На вход программе подается строка текста.

`Формат выходных данных`

Программа должна вывести кодовые значения символов строки разделенных одним символом пробела.

In [34]:
s = input()
for i in range(len(s)):
    print(ord(s[i]), end=' ')

Hello world!
72 101 108 108 111 32 119 111 114 108 100 33 

# 61. Шифр Цезаря 🌶️
Легион Цезаря, созданный в 23 веке на основе Римской Империи не изменяет древним традициям и использует шифр Цезаря. Это их и подвело, ведь данный шифр очень простой. Однако в постапокалипсисе люди плохо знают все тонкости довоенного мира, поэтому ученые из НКР не могут понять как именно нужно декодировать данные сообщения. Напишите программу для декодирования этого шифра.

`Формат входных данных`

В первой строке дается число n (1≤ n≤ 25) – сдвиг, во второй строке даётся закодированное сообщение в виде строки со строчными латинскими буквами.

`Формат выходных данных`

Программа должна вывести одну строку – декодированное сообщение. Обратите внимание, что нужно декодировать сообщение, а не закодировать.

In [35]:
n = int(input())
s = input()
for i in range(len(s)):
    dec = ord(s[i]) - n
    if dec < 97:
        dec += 26
    print(chr(dec), end='')

1
fsfftsfufksttskskt
ereesretejrssrjrjs